code/modules/admin/view_variables/debug_variable_appearance.dm

```diff
@@ -0,0 +1,201 @@
+/* 	< OH MY GOD. Can't you just make "/image/proc/foo()" instead of making these? >
+ * 		/appearance is a hardcoded byond type, and it is very internal type.
+ *		Its type is actually /image, but it isn't truly /image. We defined it as "/appearance"
+ * 		new procs to /image will only work to actual /image references, but...
+ * 		/appearance references are not capable of executing procs, because these are not real /image
+ * 		This is why these global procs exist. Welcome to the curse.
+ */
+/// Makes a var list of /appearance type actually uses. This will be only called once.
+/proc/build_virtual_appearance_vars()
+	. = list("vis_flags") // manual listing.
+	var/list/unused_var_names = list(
+		"appearance", // it only does self-reference
+		"x","y","z", // these are always 0
+		"weak_reference", // it's not a good idea to make a weak_ref on this, and this won't have it
+		"vars", // inherited from /image, but /appearance hasn't this
+
+		// Even if these vars are essential for image, these only exists in an actual type
+		"filter_data", 
+		"realized_overlays",
+		"realized_underlays",
+				
+		// we have no reason to show these, right?
+		"_active_timers",
+		"_datum_components",
+		"_listen_lookup",
+		"_signal_procs",
+		"__auxtools_weakref_id",
+		"_status_traits",
+		"cooldowns",
+		"datum_flags",
+		"visibility",
+		"verbs",
+		"gc_destroyed",
+		"harddel_deets_dumped",
+		"open_uis",
+		"tgui_shared_states"
+		)
+	var/image/dummy_image = image(null, null)
+	for(var/each in dummy_image.vars) // try to inherit var list from /image
+		if(each in unused_var_names)
+			continue
+		. += each
+	del(dummy_image)
+	dummy_image = null
+
+/// appearance type needs a manual var referencing because it doesn't have "vars" variable internally.
+/// There's no way doing this in a fancier way.
+/proc/debug_variable_appearance(var_name, appearance)
+	var/value
+	try
```

Comment:
do we really need `try/catch` here? i would rather eat a potential runtime so we know if stuff is screwed up. also maybe `NAMEOF()` would work in this switch chain?

Reply:
It doesn't throw runtime. It handles internally itself unless TG added something at `/exception/New` for that.
Also, it is not possible to use NAMEOF() in this switch chain, because we should use **string** to refer a variable. This is the curse.

Reply:
(I commented a wrong answer if you read it already)
/appearance is not trustful because it has /atom/movable values that are not documented, and vis_flags doesn't exist even if DM document says it exists. I'd rather keep this way instead of throwing runtime whenever it's investigated.

Also, it is not possible to use NAMEOF() in this switch chain, because we should use **string** to refer a variable. This is the curse.

Reply:
Hmm, it is possible to use a switch if..

```DM
#define APPEARANCE_VAR(thing, varname) (#varname, NAMEOF(thing, varname)
switch(var_name)
   if(APPEARANCE_VAR(appearance, datum_flags))
```

Reply:
Leaving it unsolved
* Switch is solved
* try/catch is not solved, but I believe it should be here.